brush-core/src/builtins/mapfile.rs

```diff
@@ -66,14 +61,20 @@ impl builtins::Command for MapFileCommand {
         // Read!
         let results = self.read_entries(input_file)?;
 
-        // Assign!
-        context.shell.env.update_or_add(
-            &self.array_var_name,
-            variables::ShellValueLiteral::Array(results),
-            |_| Ok(()),
-            env::EnvironmentLookup::Anywhere,
-            env::EnvironmentScope::Global,
-        )?;
+        for (elem_idx, result) in results.0.into_iter().enumerate() {
```

Comment:
When `-O` isn't used and the array is sufficiently large, a single `.update_or_add()` call may be more efficient, but I do like the simplicity of having a common code path for the `-O` case and non-`-O` case.

Assuming the semantics work out for all the cases we care about, I'm okay with this approach.  We can always revisit down the road.